### ISSUE TITLE
Doc comments with code at beginning includes all of code in doc output

### ISSUE THREAD

**Issue by Munksgaard** [rust-lang/rust#54975]
Putting code at the top of a docstring results in the entire code bit being included in the short summary of the docstring.

The following code

```rust
/// ```
/// foo
/// bar
/// ```
pub struct Foo;
```

Results in the following foo/index.html page

![bla](https://user-images.githubusercontent.com/230613/46764996-380d4580-ccde-11e8-98a6-b4dcff5f50b9.png)

Having large code samples there generates ugly documentation. For instance, I ran across the following while working through #54824 (in [src/test/rustdoc/issue-47183.rs](https://github.com/rust-lang/rust/blob/5af0bb83044c3fb43d59ad5f49ee5ae1a293901f/src/test/rustdoc/issue-41783.rs))

![bla2](https://user-images.githubusercontent.com/230613/46765035-53785080-ccde-11e8-8e2c-c31f417b366b.png)

Which is generated by the following code:

```
/// ```no_run
/// # # space
/// # comment
/// ## single
/// ### double
/// #### triple
/// ##[outer]
/// ##![inner]
/// ```
pub struct Foo;
```

My suggestion would be that rustdoc should detect whether the first line of a doc comment is the beginning of a code block, and if so, it should leave it out of the "preview".

**Comment by GuillaumeGomez** [comment]
Makes sense. I'll send a patch tomorrow.
